index.js

```diff
@@ -189,55 +204,65 @@ EventEmitter.prototype.on = function on(event, fn, context) {
  * @param {String|Symbol} event The event name.
  * @param {Function} fn The listener function.
  * @param {Mixed} [context=this] The context to invoke the listener with.
- * @returns {EventEmitter} `this`.
+ * @returns {EventEmitter} Reference to the EventEmitter instance.
  * @api public
  */
 EventEmitter.prototype.once = function once(event, fn, context) {
-  var listener = new EE(fn, context || this, true)
-    , evt = prefix ? prefix + event : event;
-
-  if (!this._events[evt]) this._events[evt] = listener, this._eventsCount++;
-  else if (!this._events[evt].fn) this._events[evt].push(listener);
-  else this._events[evt] = [this._events[evt], listener];
-
-  return this;
+  return _addListener(this, event, fn, context, true);
 };
 
 /**
+ * Clear event by name
+ *
+ * @param {EventEmitter} emitter Reference to the EventEmitter instance.
+ * @param {String|Symbol} evt The Event name
+ * @api private
+ */
+function clearEvent(emitter, evt) {
+  if (--emitter._eventsCount === 0) emitter._events = new Events();
+  else delete emitter._events[evt];
+}
+
+/**
  * Remove the listeners of a given event.
  *
- * @param {String|Symbol} event The event name.
+ * @param {String|Symbol} [event] The event name.
  * @param {Function} fn Only remove the listeners that match this function.
  * @param {Mixed} context Only remove the listeners that have this context.
  * @param {Boolean} once Only remove one-time listeners.
- * @returns {EventEmitter} `this`.
+ * @returns {EventEmitter} Reference to the EventEmitter instance.
  * @api public
  */
 EventEmitter.prototype.removeListener = function removeListener(event, fn, context, once) {
-  var evt = prefix ? prefix + event : event;
+  var evt;
 
+  if (!event) {
```

Comment:
This is a breaking change.

```js
ee.removeListener();
```

does not remove all events when no event is specified.